### PR TITLE
Bug 1315663 - Add engagement scalars

### PR DESCRIPTION
**Pull request by mreid-moz**
This change is [Reviewable](https://reviewable.io/reviews/mozilla/telemetry-batch-view/147)

### CODE REVIEW

File: src/main/scala/com/mozilla/telemetry/views/MainSummaryView.scala [hunk 1]
```diff
@@ -518,6 +536,20 @@ object MainSummaryView {
     StructField("dom_ipc_process_count", IntegerType, nullable = true) // dom.ipc.processCount
   ))
 
+  // Data for browser engagement measures, taken from:
+  //  processes.parent.scalars["browser.engagement.*"]
+  // For more information, see the Scalars definitions at
+  //  https://dxr.mozilla.org/mozilla-central/source/toolkit/components/telemetry/Scalars.yaml
+  def buildBrowserEngagementSchema = StructType(List(
+    StructField("max_concurrent_tab_count", IntegerType, nullable = true),
```
Comment (vitillo) [inline src/main/scala/com/mozilla/telemetry/views/MainSummaryView.scala]: Are we going to have to back-fill if a new field is added in the future?

Comment (mreid-moz) [inline src/main/scala/com/mozilla/telemetry/views/MainSummaryView.scala]: I'm just checking the behaviour of adding a new field within a nested struct now.

If it turns out that it _would_ require backfill, I guess we should modify these to be top-level fields?

Comment (vitillo) [inline src/main/scala/com/mozilla/telemetry/views/MainSummaryView.scala]: Do you have a notebook with the tests so we can re-use it next time we have the same doubt?

Comment (mreid-moz) [inline src/main/scala/com/mozilla/telemetry/views/MainSummaryView.scala]: Things work fine in Spark when adding a new field to a struct, as long as the data is loaded using `sqlContext.read.option("mergeSchema", "true").load(dataset, 'parquet')`.

I'm not too sure how to test prestodb, I may need to invoke @fbertsch :)

Comment (fbertsch) [inline src/main/scala/com/mozilla/telemetry/views/MainSummaryView.scala]: @mreid-moz it's not pretty but you could submit a PR to https://github.com/fbertsch/schema_evolution_exploration with the scenario you want to test, and I'll run it on a presto cluster at some point.

Comment (vitillo) [inline src/main/scala/com/mozilla/telemetry/views/MainSummaryView.scala]: Let's check if Presto supports this and if that's the case then let's time some Spark main_summary queries with `mergeSchema=true` to see if that's an issue at all.

Comment (mreid-moz) [inline src/main/scala/com/mozilla/telemetry/views/MainSummaryView.scala]: D'oh.

OK, looks like separate columns is the way to go 😭

**Comment by mreid-moz** [comment]
@vitillo I've refactored to use simple columns instead of a struct. Please take another look.

---

Review status: 0 of 5 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/147#-:-KXll9Szs-9OvMeW_ski:b-bjmfqn)*

**Comment by vitillo** [comment]
Please merge the first two commits into a single one.

---

Review status: 0 of 5 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/147#-:-KXlxa9ayv7cUCXdb2yw:b53hbdd)*

**Comment by vitillo** [comment]
Review status: 0 of 5 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/test/scala/com/mozilla/telemetry/MainSummaryViewTest.scala, line 886 at r4](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/147#-KXlyend9NB1aTVJhceI:-KXlyend9NB1aTVJhceJ:b-ksqmgb) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/452e7b485f3a0fb9667d6934d4b3cb19ebcb732f/src/test/scala/com/mozilla/telemetry/MainSummaryViewTest.scala#L886)):*
> ```Scala
>     // submitted with a URL of
>     //    /submit/telemetry/bar/main/Firefox/53.0a1/nightly/20161117030212
>     val hekaFileName = "/test_main.2.snappy.heka"
> ```

Where is `test_main.2.json.gz` used?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/147)*

**Comment by vitillo** [comment]
Review status: 0 of 5 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/test/scala/com/mozilla/telemetry/MainSummaryViewTest.scala, line 886 at r4](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/147#-KXlyend9NB1aTVJhceI:-KXlynSKeW5Mz6sZbJNM:b3od2e2) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/452e7b485f3a0fb9667d6934d4b3cb19ebcb732f/src/test/scala/com/mozilla/telemetry/MainSummaryViewTest.scala#L886)):*
<details><summary><i>Previously, vitillo (Roberto Agostino Vitillo) wrote…</i></summary><blockquote>

Where is `test_main.2.json.gz` used?
</blockquote></details>

How is that file generated? What's the procedure to update it?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/147)*

**Comment by mreid-moz** [comment]
Review status: 0 of 5 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/test/scala/com/mozilla/telemetry/MainSummaryViewTest.scala, line 886 at r4](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/147#-KXlyend9NB1aTVJhceI:-KXpSD2_1tWaHqYRHE6l:b-y9a894) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/452e7b485f3a0fb9667d6934d4b3cb19ebcb732f/src/test/scala/com/mozilla/telemetry/MainSummaryViewTest.scala#L886)):*
<details><summary><i>Previously, vitillo (Roberto Agostino Vitillo) wrote…</i></summary><blockquote>

How is that file generated? What's the procedure to update it?
</blockquote></details>

`test_main.2.json.gz` is not directly used, it's included as a convenience for looking at the contents of the record in its raw json form (as well as making things easier if one wanted to update the heka version - you could edit the json and re-convert).

The framed-heka version is generated by running the json doc through the ingestion pipeline and saving it to file locally (using `curl http://localhost/<submission url in comment above> --data-binary @test_main.2.json.gz`)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/147)*

**Comment by vitillo** [comment]
Review status: 0 of 5 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/test/scala/com/mozilla/telemetry/MainSummaryViewTest.scala, line 886 at r4](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/147#-KXlyend9NB1aTVJhceI:-KXpZRfoRjXB9SzkXjHO:b-4genyo) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/452e7b485f3a0fb9667d6934d4b3cb19ebcb732f/src/test/scala/com/mozilla/telemetry/MainSummaryViewTest.scala#L886)):*
<details><summary><i>Previously, mreid-moz (Mark Reid) wrote…</i></summary><blockquote>

`test_main.2.json.gz` is not directly used, it's included as a convenience for looking at the contents of the record in its raw json form (as well as making things easier if one wanted to update the heka version - you could edit the json and re-convert).

The framed-heka version is generated by running the json doc through the ingestion pipeline and saving it to file locally (using `curl http://localhost/<submission url in comment above> --data-binary @test_main.2.json.gz`)
</blockquote></details>

This has to be all documented somewhere so we know how to regenerate each one of those artifact if needed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/147)*

**Comment by vitillo** [comment]
Review status: 0 of 5 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/test/scala/com/mozilla/telemetry/MainSummaryViewTest.scala, line 886 at r4](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/147#-KXlyend9NB1aTVJhceI:-KXpZuGoHAud6W-CltcV:bgu5161) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/452e7b485f3a0fb9667d6934d4b3cb19ebcb732f/src/test/scala/com/mozilla/telemetry/MainSummaryViewTest.scala#L886)):*
<details><summary><i>Previously, vitillo (Roberto Agostino Vitillo) wrote…</i></summary><blockquote>

This has to be all documented somewhere so we know how to regenerate each one of those artifact if needed.
</blockquote></details>

Even better if we have a script to create an updated version of both.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/147)*

**Comment by mreid-moz** [comment]
Review status: 0 of 5 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/test/scala/com/mozilla/telemetry/MainSummaryViewTest.scala, line 886 at r4](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/147#-KXlyend9NB1aTVJhceI:-KXp_Hgp3filCVj9kuXU:b-jox9cl) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/452e7b485f3a0fb9667d6934d4b3cb19ebcb732f/src/test/scala/com/mozilla/telemetry/MainSummaryViewTest.scala#L886)):*
<details><summary><i>Previously, vitillo (Roberto Agostino Vitillo) wrote…</i></summary><blockquote>

Even better if we have a script to create an updated version of both.
</blockquote></details>

It is non-trivial to document how to run the ingestion pipeline, let alone make it scriptable. I am working towards that as part of another project, but I think it's beyond the scope of this PR.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/147)*

**Comment by vitillo** [comment]
Review status: 0 of 5 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/test/scala/com/mozilla/telemetry/MainSummaryViewTest.scala, line 886 at r4](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/147#-KXlyend9NB1aTVJhceI:-KXpg8nZJb50kVjlDwnB:b-lzu5d6) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/452e7b485f3a0fb9667d6934d4b3cb19ebcb732f/src/test/scala/com/mozilla/telemetry/MainSummaryViewTest.scala#L886)):*
<details><summary><i>Previously, mreid-moz (Mark Reid) wrote…</i></summary><blockquote>

It is non-trivial to document how to run the ingestion pipeline, let alone make it scriptable. I am working towards that as part of another project, but I think it's beyond the scope of this PR.
</blockquote></details>

I understand but I can't let this in like that without knowing how to generate those artifacts. Feel free to remove them for now, even if it reduces the test coverage.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/147)*

**Comment by mreid-moz** [comment]
Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/test/scala/com/mozilla/telemetry/MainSummaryViewTest.scala, line 886 at r4](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/147#-KXlyend9NB1aTVJhceI:-KXpjvDYxFAMAU6dSNKp:b-896fix) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/452e7b485f3a0fb9667d6934d4b3cb19ebcb732f/src/test/scala/com/mozilla/telemetry/MainSummaryViewTest.scala#L886)):*
<details><summary><i>Previously, vitillo (Roberto Agostino Vitillo) wrote…</i></summary><blockquote>

I understand but I can't let this in like that without knowing how to generate those artifacts. Feel free to remove them for now, even if it reduces the test coverage.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/147)*

**Comment by vitillo** [comment]
:lgtm:

---

Reviewed 2 of 3 files at r3, 3 of 3 files at r5.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/147#-:-KXpoih3kCKSCvAHi59R:bnfp4nl)*

**Pull request merged**